src/Parallel/Algorithms/AlgorithmArray.hpp

```diff
@@ -0,0 +1,59 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// This file should be included in each file which defines an array
+/// parallel component; doing so ensures that the correct charm++ chares are
+/// defined for executables that use that parallel component.
+
+#pragma once
+
+#include "Parallel/Algorithm.hpp"
+#include "Parallel/Algorithms/AlgorithmArrayDeclarations.hpp"
+#include "Parallel/ArrayIndex.hpp"
+
+/*!
+ * \ingroup ParallelGroup
+ * \brief A charm++ algorithm array chare.
```

Comment:
[nitpick] "algorithm" is a spectre concept, so I would un-mix it from the charm concepts by saying "An algorithm that wraps a charm++ array chare" or similar. What do you think? It's certainly possible I'm not understanding something well enough 🙃 

(Same comment for the other algorithm types)